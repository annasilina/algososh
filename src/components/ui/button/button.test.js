import React from 'react';
import renderer from 'react-test-renderer';

import {Button} from './button';

describe('button', () => {
	it('with text "Button" rendered correctly', () => {
		const element = renderer
			.create(<Button text='Button'/>)
			.toJSON();
		expect(element).toMatchSnapshot();
	});

	it('without text rendered correctly', () => {
		const element = renderer
			.create(<Button/>)
			.toJSON();
		expect(element).toMatchSnapshot();
	});

	it('disabled rendered correctly', () => {
		const element = renderer
			.create(<Button disabled={true}/>)
			.toJSON();
		expect(element).toMatchSnapshot();
	});

	it('with loader rendered correctly', () => {
		const element = renderer
			.create(<Button isLoader={true}/>)
			.toJSON();
		expect(element).toMatchSnapshot();
	});
});