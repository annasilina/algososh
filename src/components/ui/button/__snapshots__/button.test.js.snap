// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button disabled rendered correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`button with loader rendered correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`button with text "Button" rendered correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Button
  </p>
</button>
`;

exports[`button without text rendered correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
